### PR TITLE
CMake: fix config-dependent compiler flags.

### PR DESCRIPTION
**Pull request by tamaskenez**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -22,20 +22,13 @@ find_package(Threads QUIET REQUIRED)
...
 else()
 	# Ignore Visual Studio pragma regions
-	set(CMAKE_CXX_FLAGS "-Wno-unknown-pragmas")
-	if(CMAKE_BUILD_TYPE MATCHES Debug)
-		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3")
-	else()
-		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Og -g")
-	endif()
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-pragmas")
+  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Og")
```
Comment (tamaskenez) [inline CMakeLists.txt]: `-g` for debug and `-O3` for release configs are added automatically by CMake.

**Comment by tamaskenez** [comment]
Q: Don't you need `/W4` for all configurations?

**Comment by vincentlaucsb** [comment]
> Q: Don't you need /W4 for all configurations?

Yeah that'd be preferable. I forgot exactly why I excluded it from all releases. Maybe it's a relic of the past when compiling this library produced lots of warnings.

**Comment by vincentlaucsb** [comment]
Thanks again for your contributions!

**Pull request merged**